### PR TITLE
Template API: Added method for editing device or CLI template

### PR DESCRIPTION
**Pull request by ravindra-h**
- Added method for editing device or CLI template
- Enhanced attach method to support re-attaching of device / CLI template to device(s) after editing

# Pull Request summary:
Template API: Added method for editing device or CLI template

# Description of changes:
- Added method for editing device or CLI template
- Enhanced attach method to support re-attaching of device / CLI template to device(s) after editing

# Checklist:
- [x] Make sure to run pre-commit before commiting changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: vmngclient/api/template_api.py [hunk 1]
```diff
@@ -135,6 +173,8 @@ def attach(self, name: str, device: Device) -> bool:
                 }
             ]
         }
+        if is_edited:
+            payload["deviceTemplateList"][0]["isEdited"] = True  # type: ignore
```
Comment (sbasan) [inline vmngclient/api/template_api.py:177]: isn't that better?
```python
payload = {
    "deviceTemplateList": [
        {
            "templateId": template_id,
            "isEdited": is_edited,
            "device": [
                {
                    "csv-status": "complete",
                    "csv-deviceId": device.uuid,
                    "csv-deviceIP": device.id,
                    "csv-host-name": device.hostname,
                    "csv-templateId": template_id,
                }
            ],
        }
    ]
}
```

Comment (sbasan) [reply vmngclient/api/template_api.py:177]: This is what I suspected, however are we sure that "isEdited" = False is different from "isEdited" missing in request to vManage?

**Review by JimOverholt — COMMENTED** [review]
I know you want to get this merge.
Check if we need to follow up and add a unit test

**Review by JimOverholt — APPROVED** [review]

**Comment by ravindra-h** [comment]
> I know you want to get this merge. Check if we need to follow up and add a unit test

Hi Jim,

I'll try to add UT for this by next week. Meanwhile if I can get any documentation / TOI on adding UT it would be helpful.

**Comment by ravindra-h** [comment]
@sbasan I've responded to your PR comments. Please check and let me know if there are any further queries, else request to approve the PR.

**Review by sbasan — APPROVED** [review]
lgtm

**Pull request merged**